quadro.c

```diff
@@ -1,52 +1,104 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <math.h>
+#include <stdlib.h>
 
-bool sign_of_diskr(double a, double b, double c){
-    double D = (b*b - 4*a*c);
-    if (D>0)
+struct coef
+{
+    double a;
+    double b;
+    double c;
+};
+
+struct roots
+{
+    double* root1;
+    double* root2;
+};
+
+struct coef quadro_input() {
+    struct coef coefs;
+    coefs.a = 0;
+    coefs.b = 0;
+    coefs.c = 0;
+    printf("Enter coefficient a: ");
+    while (scanf("%lf", &coefs.a) != 1) {
+        printf("Error input for a, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    printf("Enter coefficient b: ");
+    while (scanf("%lf", &coefs.b) != 1) {
+        printf("Error input for b, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    printf("Enter coefficient c: ");
+    while (scanf("%lf", &coefs.c) != 1) {
+        printf("Error input for c, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    return coefs;
+}
+
+bool sign_of_diskr(double d){
+    if (d>0)
         return(true);
     else
         return(false);
 }
 
 double diskr(double a, double b, double c){
-    double D = (b*b - 4*a*c);
-    return (D);
+    return (b*b - 4*a*c);
+}
+
+struct roots quadro_output(double a, double b, double c) {
+    struct roots roots;
+    roots.root1 = NULL;
+    roots.root2 = NULL;
+
+    if (a == 0) {
+        if (b != 0) {
+            roots.root1 = (double*)malloc(sizeof(double));
+            *roots.root1 = -c / b;
+        }
+        return roots;
+    }
+
+    double d = diskr(a, b, c);
+    if (sign_of_diskr(d)) {
+        roots.root1 = (double*)malloc(sizeof(double));
+        roots.root2 = (double*)malloc(sizeof(double));
+        *roots.root1 = (-b + sqrt(d)) / (2 * a);
+        *roots.root2 = (-b - sqrt(d)) / (2 * a);
+    } else if (d == 0) {
+        roots.root1 = (double*)malloc(sizeof(double));
+        *roots.root1 = -b / (2 * a);
+    }
+
+    return roots;
 }
 
 int main(){
-    double a, b, c;
-    printf("vvedite a");
-    scanf("%lf", &a);
-    printf("vvedite b");
-    scanf("%lf", &b);
-    printf("vvedite c");
-    scanf("%lf", &c);
-
-    if (a==0){
-        printf("ne kvadratnoe");
+    struct coef coefs = quadro_input();
+    struct roots roots = quadro_output(coefs.a, coefs.b, coefs.c);
```

Comment:
you've named function which solves quadratic equation as `quadro_output`, and then printed your solution. where is a naming mistake?